docker/lib/dependabot/docker/file_updater.rb

```diff
@@ -109,46 +121,55 @@ def update_digest_and_tag(previous_content, old_source, new_source)
         end
       end
 
+      sig { params(source: T::Hash[Symbol, T.nilable(String)]).returns(T::Boolean) }
       def specified_with_tag?(source)
-        source[:tag]
+        !!source[:tag]
       end
 
+      sig { params(source: T::Hash[Symbol, T.nilable(String)]).returns(T::Boolean) }
       def specified_with_digest?(source)
-        source[:digest]
+        !!source[:digest]
       end
 
+      sig { params(file: Dependabot::DependencyFile).returns(T::Array[String]) }
       def new_tags(file)
         requirements(file)
           .map { |r| r.fetch(:source)[:tag] }
       end
 
+      sig { params(file: Dependabot::DependencyFile).returns(T::Array[String]) }
       def old_tags(file)
-        previous_requirements(file)
-          .map { |r| r.fetch(:source)[:tag] }
+        T.must(previous_requirements(file))
+         .map { |r| r.fetch(:source)[:tag] }
       end
 
+      sig { params(source: T::Hash[Symbol, T.nilable(String)]).returns(T.nilable(String)) }
       def private_registry_url(source)
         source[:registry]
       end
 
+      sig { params(file: Dependabot::DependencyFile).returns(T::Array[T::Hash[Symbol, T.nilable(String)]]) }
       def sources(file)
         requirements(file).map { |r| r.fetch(:source) }
       end
 
+      sig { params(file: Dependabot::DependencyFile).returns(T::Array[T::Hash[Symbol, T.nilable(String)]]) }
       def previous_sources(file)
-        previous_requirements(file).map { |r| r.fetch(:source) }
+        T.must(previous_requirements(file)).map { |r| r.fetch(:source) }
       end
 
+      sig { params(file: Dependabot::DependencyFile).returns(String) }
       def updated_yaml_content(file)
         updated_content = file.content
-        updated_content = update_helm(file, updated_content) if Utils.likely_helm_chart?(file)
-        updated_content = update_image(file, updated_content)
+        updated_content = update_helm(file, T.must(updated_content)) if Utils.likely_helm_chart?(file)
+        updated_content = update_image(file, T.must(updated_content))
```

Comment:
Not sure this is the correct approach here. `file.content` can be `nil`. Can you try and allow the signatures of the called methods to accept that?